### PR TITLE
Made clear error messages in Source File Relocator

### PR DESCRIPTION
**Pull request by c5h11oh**
When the relocator finds file(s) that are in scan folder but are absent in the asset database, proper error messages are emitted.

Signed-off-by: Sven Hwang <30003118+c5h11oh@users.noreply.github.com>

### CODE REVIEW

File: Code/Tools/AssetProcessor/native/AssetManager/SourceFileRelocator.cpp [hunk 1]
```diff
@@ -387,12 +397,15 @@ Please note that only those seed files will get updated that are active for your
...
 
-                GetFilesFromSourceControl(sources, scanFolderInfoOut, normalizedSource.c_str(), excludeMetaDataFiles);
+                fileExists = GetFilesFromSourceControl(sources, scanFolderInfoOut, normalizedSource.c_str(), excludeMetaDataFiles);
             }
 
             if(sources.empty())
             {
-                return AZ::Failure(AZStd::string("Wildcard search did not match any files.\n"));
+                if (fileExists)
+                    return AZ::Failure(AZStd::string("Wildcard search did not match any files found in the asset database.\n"));
```
Comment (jackalbe) [inline Code/Tools/AssetProcessor/native/AssetManager/SourceFileRelocator.cpp:406]: is there any way to include the file pattern or any input that might have caused this to fail?

File: Code/Tools/AssetProcessor/native/AssetManager/SourceFileRelocator.cpp [hunk 2]
```diff
@@ -413,11 +426,14 @@ Please note that only those seed files will get updated that are active for your
...
             }
 
-            GetFilesFromSourceControl(sources, scanFolderInfoOut, absoluteSourcePath.c_str(), excludeMetaDataFiles);
+            fileExists = GetFilesFromSourceControl(sources, scanFolderInfoOut, absoluteSourcePath.c_str(), excludeMetaDataFiles);
 
             if (sources.empty())
             {
-                return AZ::Failure(AZStd::string("File not found.\n"));
+                if (fileExists)
+                    return AZ::Failure(AZStd::string("File is not found in the asset database and will not be processed.\n"));
```
Comment (jackalbe) [inline Code/Tools/AssetProcessor/native/AssetManager/SourceFileRelocator.cpp:434]: is there any way to include the file pattern or any input that might have caused this to fail?

File: Code/Tools/AssetProcessor/native/AssetManager/SourceFileRelocator.cpp (hunk 1 above)
Comment (jackalbe) [reply Code/Tools/AssetProcessor/native/AssetManager/SourceFileRelocator.cpp]: Also the code standard is to include curly braces for each logical expression:
if (foo)
{
// do logic
}
https://github.com/o3de/sig-core/blob/main/governance/Coding-Standards-and-Style-Guide.md#use-of-curly-braces

**Review by jackalbe — APPROVED** [review]

**Review by c5h11oh — COMMENTED** [review]
reply the reviews

**Review by amzn-mike — APPROVED** [review]

File: Code/Tools/AssetProcessor/native/AssetManager/SourceFileRelocator.cpp [hunk 3]
```diff
@@ -167,19 +167,27 @@ Please note that only those seed files will get updated that are active for your
...
                     sources.emplace_back(entry, GetProductMapForSource(entry.m_sourceID), entry.m_sourceName, scanFolderInfo);
+                    filesNotInAssetDatabase.remove(databaseSourceName);
                     sourceIndexMap[databaseSourceName] = aznumeric_cast<int> (sources.size() - 1);
                     return true;
                 });
         }
 
+        for (const QString& file : filesNotInAssetDatabase)
+        {
+            AZ_Printf("AssetProcessor", "File `%s` was found/matched but is not a source asset.  Skipping.\n", file.toUtf8().constData());
```
Comment (lmbr-pip) [inline Code/Tools/AssetProcessor/native/AssetManager/SourceFileRelocator.cpp:188]: Am slightly confused by output, surely file is a "source asset" ie its an asset on disc, its just that we do not have a mapping in the asset database (ie it failed to process?)

Should this be more friendly to user?
```suggestion
            AZ_Printf("AssetProcessor", "File `%s` matched during relocation does not have entry in Asset database.  Skipping.\n", file.toUtf8().constData());
```

Do we want to add any information about how to fix? ie

```
if  (!filesNotInassetDatabase.empty(){
    AZ_Printf("AssetProcessor", "The following files were found without entries in the asset database. Examine AP logs to identify reasons for why asset was ignored or failed to process. "
}
```

**Review by lmbr-pip — COMMENTED** [review]
LG.

Please add how tested to PR description. It may also be useful to show output of logs statements if possible.

File: Code/Tools/AssetProcessor/native/AssetManager/SourceFileRelocator.cpp (hunk 3 above)
Comment (c5h11oh) [reply Code/Tools/AssetProcessor/native/AssetManager/SourceFileRelocator.cpp:188]: I am not sure users conceive a source asset as an entry in the asset database, or a file on disk. Your suggestion is probably better, as it clearly points out the problem is "no entry in the database".
However, I feel a bit confused by a file being "matched" but "does not have entry in the Asset DB". Should we keep the wording "found" such that it is clear that the file "exists on disk"?
e.g. does this sound better to you? "File `%s` founded on disk but does not have an entry in Asset database.  Skipping this relocation.\n"

Comment (c5h11oh) [reply Code/Tools/AssetProcessor/native/AssetManager/SourceFileRelocator.cpp:188]: Current output example:
```
AssetProcessor: File `Assets/Test10641/test - Copy (3).mtl` was found/matched but is not a source asset.  Skipping.
AssetProcessor: File `Assets/Test10641/test - Copy (2).mtl` was found/matched but is not a source asset.  Skipping.
AssetProcessor: File `Assets/Test10641/test.mtl` was found/matched but is not a source asset.  Skipping.
AssetProcessor: File `Assets/Test10641/test - Copy.mtl` was found/matched but is not a source asset.  Skipping.
AssetProcessor: Wildcard search matched one or more files but none are source assets.  This utility only handles source assets.
```

**Pull request merged**